column_chunk.go

```diff
@@ -23,8 +23,8 @@ type ColumnChunk interface {
 	// methods, programs must treat those as read-only.
 	//
 	// If the column chunk does not have a page index, the methods return nil.
-	ColumnIndex() ColumnIndex
-	OffsetIndex() OffsetIndex
+	ColumnIndex() (ColumnIndex, error)
+	OffsetIndex() (OffsetIndex, error)
```

Comment:
It's a bit unfortunate that we have to introduce this breaking change, but I agree that the benefits outweigh the cost 👍 

Could you add documentation to explain why the breaking change was introduced, and how applications are expected to adapt? This will go a long way in helping users upgrade their applications. Something like _"Prior to v0.20, these methods did not return an error because etc..."_

Reply:
Good idea, I added additional docs on these methods.

I noticed that we do not keep the CHANGELOG up to date though. Maybe we want to require an entry for each PR under an `Upcoming` section, so that whoever cuts a new release would not have to review everything prior to the last release.

Reply:
So far I think we've only used the auto-generated changelog that github produces in each release; those tend to be heavy weight time investment that personally I cannot allocate for now, so an ad-hoc option when there is a breaking change is a bandaid for it :) So far those have been rare, I'm hoping we can keep it this way 🤞